src/js/controller/FramesListController.js

```diff
@@ -87,7 +87,11 @@
     } else if (action === ACTION.SELECT) {
       this.piskelController.setCurrentFrameIndex(index);
     } else if (action === ACTION.NEW_FRAME) {
-      this.piskelController.addFrame();
+      // Publish an event that the user selected add new frame.
+      $.publish(Events.ADD_NEW_FRAME_CLICKED);
+      // Disable adding a blank frame and idead use the PiskelApi
+      // to cause an action.
+      //this.piskelController.addFrame();
```

Comment:
A couple of alternatives:

* This could do the old behavior if no custom `newFrameClicked` handler has been attached.
* We could do more refactoring here to make the behavior more configurable - perhaps have a default controller that normally listens for and handles `ADD_NEW_FRAME_CLICKED` and then skip loading that controller and have our own controller handle it in our particular use case.

Both are probably overkill for now though.